llpc/tool/llpcCompilationUtils.cpp

```diff
@@ -328,22 +328,23 @@ Error buildShaderModules(ICompiler *compiler, CompileInfo *compileInfo) {
 // =====================================================================================================================
 // Output LLPC resulting binary (ELF binary, ISA assembly text, or LLVM bitcode) to the specified target file.
 //
-// @param compileInfo : Compilation info of LLPC standalone tool
+// @param pipelineBin : Output elf pipeline binary
 // @param suppliedOutFile : Name of the file to output ELF binary (specify "" to use base name of first input file with
 // appropriate extension; specify "-" to use stdout)
 // @param firstInFile : Name of first input file
 // @returns : `ErrorSuccess` on success, `ResultError` on failure
-Error outputElf(CompileInfo *compileInfo, const std::string &suppliedOutFile, StringRef firstInFile) {
-  const BinaryData &pipelineBin = (compileInfo->stageMask & ShaderStageComputeBit)
-                                      ? compileInfo->compPipelineOut.pipelineBin
-                                      : compileInfo->gfxPipelineOut.pipelineBin;
+Error outputElf(const BinaryData &pipelineBin, const std::string &suppliedOutFile, StringRef firstInFile) {
```

Comment:
I was thinking of moving the whole implementation to `PipelineBuilder`. `outputElf` only decides the filename and calls `writeFile` at the end, so I don't think it has much use outside of the builder.